pkg/cosign/verify.go

```diff
@@ -196,15 +213,54 @@ func VerifyImageSignatures(ctx context.Context, signedImgRef name.Reference, co
 
 	// TODO(mattmoor): We could implement recursive verification if we just wrapped
 	// most of the logic below here in a call to mutate.Map
-
 	se, h, err := getSignedEntity(signedImgRef, co.RegistryClientOpts)
 	if err != nil {
 		return nil, false, err
 	}
-	sigs, err := se.Signatures()
-	if err != nil {
-		return nil, false, err
+
+	var sigs oci.Signatures
+	sigRef := co.SignatureRef
+	if sigRef == "" {
+		sigs, err = se.Signatures()
+		if err != nil {
+			return nil, false, err
+		}
+	} else {
```

Comment:
nit: I'd suggest moving this piece of code to a function to reduce the size and logic in this function.

Reply:
thank you @hectorj2f, done.